checks/tools/io/cdo.py

```diff
@@ -21,60 +20,100 @@
 # 'test_hgroups.nc4'; it gives the error:
 #   cdo info: Open failed on >test_hgroups.nc4<
 #   Unsupported file structure"
-
+# }}}
 import os
 
 import reframe as rfm
 import reframe.utility.sanity as sn
 
 
 class CDOBaseTest(rfm.RunOnlyRegressionTest):
-    def __init__(self):
-        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
-                                       'CDO-NCO')
-        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu',
-                              'dom:mc', 'arolla:pn', 'tsa:pn',
-                              'eiger:mc', 'pilatus:mc']
+    valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu',
+                     'dom:mc', 'arolla:pn', 'tsa:pn',
+                     'eiger:mc', 'pilatus:mc',
+                     'manali:gpu-squashfs', 'balfrin:gpu-squashfs']
+    # TODO: update maintainers ?
+    maintainers = ['SO', 'CB']
+    tags = {'production', 'mch', 'external-resources'}
+
+    @run_after('init')
+    def setup_valid_pe(self):
+        modules = {
+            'arolla': ['cdo', 'netcdf-fortran'],
+            'tsa': ['cdo', 'netcdf-fortran'],
+            'eiger': ['CDO'],
+            'pilatus': ['CDO'],
+            'manali': ['cdo', 'netcdf-fortran'],
+            'balfrin': ['cdo', 'netcdf-fortran'],
+        }
+        if self.current_system.name in modules:
+            self.modules = modules[self.current_system.name]
+        else:
+            self.modules = ['CDO']
+
         if self.current_system.name in ['arolla', 'tsa']:
             self.exclusive_access = True
             self.valid_prog_environs = ['PrgEnv-gnu', 'PrgEnv-gnu-nompi']
-            self.modules = ['cdo', 'netcdf-fortran']
         elif self.current_system.name in ['eiger', 'pilatus']:
             self.valid_prog_environs = ['cpeGNU']
-            self.modules = ['CDO']
+        elif self.current_system.name in ['manali', 'balfrin']:
+            self.valid_prog_environs = ['PrgEnv-gnu']
+            self.squashfs_script = './exe.sh'
+            # until VCMSA-102 is resolved:
+            if self.current_system.name in ['balfrin']:
+                self.prerun_cmds += [
+                    r'echo "export LD_LIBRARY_PATH=\$LD_LIBRARY_PATH:'
```

Comment:
If you don't, you wil get:
```
export LD_LIBRARY_PATH=.::/scratch/e1000/piccinal/balfrin/lib64
```
instead of
```
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SCRATCH/balfrin/lib64
```